qui/updater.py

```diff
@@ -124,14 +126,36 @@ def populate_vm_list(self):
                 self.vm_list.add(VMListBoxRow(vm, state))
 
         for vm in self.qapp.domains:
-            if getattr(vm, 'updateable', False) and vm.klass != 'AdminVM'\
-                    or str(vm.name).startswith("devel-"):
+            if getattr(vm, 'updateable', False) and vm.klass != 'AdminVM':
                 try:
                     state = vm.features.get('updates-available', False)
                 except exc.QubesDaemonCommunicationError:
                     state = False
-                state = str(vm.name).startswith("devel-")
                 result = result or state
+
+                today = datetime.today()
+                try:
+                    last_update_str = vm.features.get(
+                        'last-update',
+                        datetime(1970, 1, 1, 0, 0, 0).strftime(
+                            '%Y-%m-%d %H:%M:%S'))
+                    last_update = datetime.fromisoformat(last_update_str)
+                    if (today - last_update).days > 14:
```

Comment:
I don't think "forcing" update just because there were none in the last 14 days is a good idea. On the other hand, if there were no updates check in last N days, it very much makes sense to force checking (but see below).